great_expectations/util.py

```diff
@@ -1899,6 +1905,37 @@ def get_context(  # noqa: PLR0913
     """
     project_config = _prepare_project_config(project_config)
 
+    if mode is not None:
```

Comment:
let's merge this parameter with the logic below to a) avoid duplication, and b) handle cases when one of the _get_***_context methods fails to return (which should raise a ValueError)